lib/ui/recommendation.js

```diff
@@ -144,9 +144,20 @@ Recommendation.prototype = {
     }
   },
   navigate: function(evt) {
-    // if it's a click, we'll have a MouseEvent; if it's a right-click, bail.
-    // else, we'll just have the (possibly null) data from the 'enter-key' event.
-    // fire a 'recommendation-navigate' event to trigger navigation.
-    // the URL should already be set in the urlbar by the time the event is received.
+    // Only navigate if it's a left click.
+    if (evt.button !== 0) {
+      return;
+    }
+
+    // Unfortunately, because the urlbar state isn't set by mousing over an
+    // item, we have to set the urlbar state, then ask it to navigate.
+    // Set state by faking a selection event...
+    const url = this.el.querySelector('#universal-search-recommendation-url').textContent;
```

Comment:
Funny, of course XUL doesn't support html5 `dataset`, so we have to use getAttribute/setAttribute. Still better than fishing out textContent.